pkg/apiserver/resthandler.go

```diff
@@ -147,7 +147,20 @@ func getRequestOptions(req *restful.Request, scope RequestScope, kind string, su
 		newQuery[subpathKey] = []string{req.PathParameter("path")}
 		query = newQuery
 	}
-	return queryToObject(query, scope, kind)
+	versioned, err := scope.Creater.New(scope.ServerAPIVersion, kind)
+	if err != nil {
+		// programmer error
+		return nil, err
+	}
+	if err := scope.Codec.DecodeParametersInto(query, versioned); err != nil {
```

Comment:
Sure. Sorry my question is not clear. I'm thinking because `DecodeParametersInto` bears a similar name as `DecodeInto`, which is capable of doing the deserialization and conversion in one step, why can't we make `DecodeParametersInto` doing that, too?

I recall there is a discussion on that Codec should only do deserialization and convertor should be explicitly called to do the conversion. Is that why you are doing desserialization and conversion here?